### ISSUE TITLE
Allow entity nametags to be visible to certain players

### ISSUE THREAD

**Issue by hkzorman** [minetest/minetest#5996]
Currently nametags can be visible to all players. What I would like to see is a way to show nametags only to certain players, e.g. those who actually care about those nametags. For example, see the two scenarios below:

- Imagine a scenario where a player can have one or many pets and can give them names, which in turn shows into nametags. Now, imagine there are lots of players with lots of pets in the same area, so this would be a huge amount of nametags, for players who don't care about others' pets nametags.
- Another scenario: a player "befriends" a NPC, and to identify so and allow the player easily find/identify his "friend" NPC, a nametag is added to the NPC. However, it becomes visible to everyone! Quite sure other players don't really care about that!

At first, I decided I would implement this myself into Minetest. I started by adding a `std::vector<std::string>` to the `Nametag` struct in `camera.h` and to other files as well (including, but didn't get to test, `l_object.cpp`. However, when I was going to implement the logic in `camera.cpp`, I noticed that `Camera::drawNametags()` I had no way to access the local player. Moreover, `drawscene.cpp` doesn't have the local player reference either. So before I started adding a pointer to `LocalPlayer` everywhere, decided to ask here first.

**Comment by paramat** [comment]
Related #4193

**Comment by paramat** [comment]
Related to #4122 as it is for the visibility of certain tags to certain players.

**Comment by paramat** [comment]
1 year bump for core dev opinions.

**Comment by Ezhh** [comment]
It would be a very nice thing on some servers.

**Comment by raymoo** [comment]
Also could be useful if you want to hide nametags from some players. For example, in a CTF game players could be allowed to see nametags of only their team members. To be robust to cheats, the server would have to not send nametags at all if it's not visible to a player, as opposed to just sending a flag not to show the nametag.

**Comment by Ezhh** [comment]
@raymoo Exactly. As admin of a survival server I have also often wished for a way to disable player names for everyone other than myself. It's useful for me to be able to track player movement, but in a pvp environment it prevents players hiding from each other if they can see the names.

**Comment by farooqkz** [comment]
We have observers API now. So this can be closed.

**Comment by SmallJoker** [comment]
The observers API controls the visibility of the entire entity. This feature request is about selectively showing the *nametag* (only).

**Comment by appgurueu** [comment]
> The observers API controls the visibility of the entire entity. This feature request is about selectively showing the nametag (only).

This effect is pretty easy to achieve by attaching a nametag-only entity. CTF does this.

And as a bonus, you get to control nametag placement better as well.

Of course, this doesn't fully "solve" the issue, as a case for a more convenient (and possibly more lightweight) feature could still be made. But it definitely reduces its priority.
